background/loadScripts.js

```diff
@@ -43,36 +44,52 @@ export default function loadScripts (port) {
     }
 
     if (!codeCache) {
-      const vendor = fetch(`http://${UI}/vendor.js`)
-        .then(checkResponseOk)
-        .then(response => response.blob());
-      const embed = fetch(`http://${UI}/embed.html`)
+      codeCacheVersion = State.version;
+
+      codeCache = fetch(`http://${UI}/embed.json`)
         .then(checkResponseOk)
-        .then(response => response.text())
-        .then(page => ({
-          styles: /styles\/embed\.([a-z0-9]{10})\.css/.exec(page),
-          scripts: /embed\.([a-z0-9]{10})\.js/.exec(page)
-        }))
-        .then(res => {
-          return Promise.all([
-            fetch(`http://${UI}/${res.styles[0]}`).then(checkResponseOk),
-            fetch(`http://${UI}/${res.scripts[0]}`).then(checkResponseOk)
-          ]);
+        .then((response) => response.json())
+        .then((embed) => {
+          const { assets } = embed;
+
+          const filteredAssets = assets
+            .filter((asset) => !/html$/.test(asset))
+            .filter((asset) => !/^embed(.+)js$/.test(asset));
+
+          const mainScript = assets.find((asset) => /^embed(.+)js$/.test(asset));
+
+          const assetsPromises = filteredAssets
+            .map((asset) => {
+              return fetch(`http://${UI}/${asset}`)
+                .then((response) => response.blob())
+                .then((blob) => {
+                  if (/\.js$/.test(asset)) {
+                    return URL.createObjectURL(blob, { type: 'application/javascript' });
+                  }
+
+                  return URL.createObjectURL(blob);
+                })
+                .then((url) => ({ path: asset, url }));
+            });
+
+          const scriptPromise = fetch(`http://${UI}/${mainScript}`)
+            .then((response) => response.text());
+
+          return Promise.all([ scriptPromise, Promise.all(assetsPromises) ]);
         })
-        .then(responses => Promise.all(responses.map(response => response.blob())));
+        .then(([ script, assets ]) => {
+          assets.forEach((asset) => {
+            const { path, url } = asset;
+            const regex = new RegExp(path, 'g');
 
-      codeCacheVersion = State.version;
-      codeCache = Promise.all([vendor, embed])
-        .then(scripts => {
-          const vendor = scripts[0];
-          const styles = scripts[1][0];
-          const embed = scripts[1][1];
-
-          // Concat blobs
-          const blob = new Blob([vendor, embed], { type: 'application/javascript' });
+            script = script.replace(regex, url);
+          });
+
+          return new Blob([ script ], { type: 'application/javascript' });
+        })
+        .then((blob) => {
           return {
             success: true,
-            styles: URL.createObjectURL(styles),
             scripts: URL.createObjectURL(blob)
```

Comment:
So it could be both ways. On the UI side, the setup would be that for the embed part, no CSS file is created. For the other part of the UI, CSS files are created, this doesn't change. The idea would be that instead of fetching two files and injecting the CSS manually, we fetch one script that would eventually inject the CSS.

Separate CSS files are great for avoiding FOUC. Here, I think it's not really needed